src/QirRuntime/public/IQuantumApi.hpp

```diff
@@ -0,0 +1,146 @@
+#pragma once
+
+#include <complex>
+
+#include "CoreTypes.hpp"
+
+namespace Microsoft
+{
+namespace Quantum
+{
+    struct IQuantumApi

```

Comment:
I think it would be quite helpful go over the IQuantumAPI and QAPI in our Friday meeting. I'll catch up outside this PR to get more alignment on the precise API before that and I'll start a spec document to capture this (of course I am happy if you want to do that, but I believe you may be eager to focus on the work with Alex :) ).